src/CacheManager.Core/BaseCacheManager.GetOrAdd.cs

```diff
@@ -20,7 +20,7 @@ public TCacheValue GetOrAdd(string key, Func<string, TCacheValue> valueFactory)
             NotNullOrWhiteSpace(key, nameof(key));
             NotNull(valueFactory, nameof(valueFactory));
 
-            return GetOrAddInternal(key, null, (k, r) => valueFactory(k));
+            return GetOrAddInternal(key, null, (k, r) => new CacheItem<TCacheValue>(k, valueFactory(k))).Value;
```

Comment:
`TryGetOrAddInternal` can indeed return null in that case, but `GetOrAddInternal` cannot? It will always throw in the described situation.